### PR TITLE
Fix e2e bug

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:
Fix e2e is still successful when there are e2e failures.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by gy95** [comment]
statefulset e2e errors:
```
1. an empty namespace may not be set during creation
2. StatefulSet.apps \"edge-statefulset-7q1ld\" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]
```

**Comment by gy95** [comment]
cc @wackxu

**Review by wackxu — APPROVED** [review]
Thanks for you fix, LGTM

File: tests/e2e/scripts/fast_test.sh [hunk 1]
```diff
@@ -36,14 +36,22 @@ then
     --image-url=nginx \
     --kube-master="https://$MASTER_IP:6443" \
     --kubeconfig=$KUBECONFIG \
-    --test.v \
-    2>&1 | tee -a /tmp/testcase.log
```
Comment (fisherxu) [inline tests/e2e/scripts/fast_test.sh:40]: Is it possible to still store the log in this file for debugging?

Comment (gy95) [reply tests/e2e/scripts/fast_test.sh:40]: can use `> /tmp/testcase.log` to replace `tee -a `, but i think this is duplicated. the log will be stored in action logs directly, we can look at action logs.

Comment (gy95) [reply tests/e2e/scripts/fast_test.sh:40]: but if we use `>`, i'm not quite sure whether the logs will be printed on action std out

Comment (gy95) [reply tests/e2e/scripts/fast_test.sh:40]: by test, if we use `>`, ginkgo will not print logs to the action stdout
cannot think a proper way to store the log in the file and use the ginkgo exit code meanwhile, so just logs to the action stdout directly

Comment (fisherxu) [reply tests/e2e/scripts/fast_test.sh:40]: If users run e2e in local host, the log file should be still needed, we can fix the e2e bug first, then explore a way to store the log file.

Comment (fisherxu) [reply tests/e2e/scripts/fast_test.sh:40]: Open an issue to track it #4174 @gy95

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**